error/s2n_errno.c

```diff
@@ -290,6 +290,13 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_INTERNAL_LIBCRYPTO_ERROR, "An internal error has occurred in the libcrypto API") \
     ERR_ENTRY(S2N_ERR_NO_RENEGOTIATION, "Only secure, server-initiated renegotiation is supported") \
     ERR_ENTRY(S2N_ERR_APP_DATA_BLOCKED, "Blocked on application data during handshake") \
+    ERR_ENTRY(S2N_ERR_KTLS_ALREADY_ENABLED, "kTLS is already enabled on the socket")  \
+    ERR_ENTRY(S2N_ERR_KTLS_MANAGED_IO, "kTLS cannot be enabled while custom I/O is configured for the connection")  \
+    ERR_ENTRY(S2N_ERR_KTLS_HANDSHAKE_NOT_COMPLETE, "kTLS can only be enabled once the handshake is completed") \
+    ERR_ENTRY(S2N_ERR_KTLS_UNSUPPORTED_PLATFORM, "kTLS is unsupported on this platform") \
+    ERR_ENTRY(S2N_ERR_KTLS_UNSUPPORTED_CONN, "kTLS is unsupported for this connection") \
+    ERR_ENTRY(S2N_ERR_KTLS_ULP, "An error occurred when attempting to configure the socket for kTLS. Ensure the 'tls' kernel module is enabled.")  \
```

Comment:
Do you think it would be better to have this be called S2N_ERR_KTLS_SOCKET?

Reply:
I wanted to be super specific in order to help customers debug better. There is going to be [another call later](https://github.com/aws/s2n-tls/pull/4071/commits/79c9e9aed5b0832e8d8d4cf197af4e9861b7486e) to set keys on the socket and that one will return a specific error `S2N_ERR_KTLS_ENABLE_CRYPTO`.

Both call `setsockopt` but users can actually do something about this error: "Ensure the 'tls' kernel module is enabled"